### ISSUE TITLE
files/REPL dialog box needs to be a modal dialog

### ISSUE THREAD

**Issue by peterhinch** [mu-editor/mu#128]
The dialog box which pops up if you click on files when the REPL is open (and vice versa) does not behave like a proper modal dialog: if you click on the main window it disappears behind the main window and the application appears to be unresponsive. For a beginner-focussed app this is confusing: the dialog box should remain in the foreground.

**Comment by peterhinch** [comment]
The problem is that a modal dialog needs a parent window otherwise it is "owned" by Qt rather thn by a specific window, So here's how to fix it (docstrings omitted for clarity)
```python
    def show_message(self, message, information=None, icon=None, parent=None):
        message_box = QMessageBox(parent)
```
and
```python
    def show_confirmation(self, message, information=None, icon=None, parent=None):
        message_box = QMessageBox(parent)
```
Then in each call provide
```python
            self._view.show_message(message, information, parent = self._view)
```

**Comment by ntoll** [comment]
Hi @peterhinch 
Great stuff!
Care to turn this into a PR? Then your contribution will be immortalised forever! :-)

**Comment by peterhinch** [comment]
I would if I weren't well out of my depth with GitHub. I'm actually working on a fork targeted at the ESP8266 running MicroPython. My [fork](https://github.com/peterhinch/mu) is forked from [eduvik's](https://github.com/eduvik/mu) which is in turn forked from your code. While I can raise PR's against eduvik I'm lost as to how to do it against your code.

**Comment by carlosperate** [comment]
When you create a pull request from your repository page it will most likely show you a comparison between your selected branch and eduvik's master. If you click on that drop down menu you should be able to select any other repository, including ntoll's mu master.

**Comment by peterhinch** [comment]
OK, I've raised a PR. I hope I've got it right - I find Python a whole lot easier than Git ;-)

**Comment by ZanderBrown** [comment]
travis isn't happy but otherwise everything appears to be in order. It does help when you reference the pull (#138) and you appear to have renamed `find_microbit()` to `find_upython_device()` and your pull doesn't reference this

**Comment by peterhinch** [comment]
This came from eduvik's fork. I clearly failed to disentangle my changes from his.

**Comment by ZanderBrown** [comment]
ah. Well would you mind if I was to setup a pull containing only your changes?

**Comment by peterhinch** [comment]
No problem.

**Comment by carlosperate** [comment]
This issue has been fixed in #141. Thanks for the contributions!
